dice_ml/explainer_interfaces/dice_genetic.py

```diff
@@ -325,30 +342,34 @@ def compute_yloss(self, cfs, desired_range, desired_class):
                 yloss = np.zeros(len(predicted_value))
                 for i in range(len(predicted_value)):
                     if not desired_range[0] <= predicted_value[i] <= desired_range[1]:
-                        yloss[i] = min(abs(predicted_value[i] - desired_range[0]), abs(predicted_value[i] - desired_range[1]))
+                        yloss[i] = min(abs(predicted_value[i] - desired_range[0]),
+                                       abs(predicted_value[i] - desired_range[1]))
             return yloss
 
     def compute_proximity_loss(self, x_hat_unnormalized, query_instance_normalized):
         """Compute weighted distance between two vectors."""
         x_hat = self.data_interface.normalize_data(x_hat_unnormalized)
-        feature_weights = np.array([self.feature_weights_list[0][i] for i in self.data_interface.continuous_feature_indexes])
-        product = np.multiply((abs(x_hat - query_instance_normalized)[:, [self.data_interface.continuous_feature_indexes]]),
-                              feature_weights)
+        feature_weights = np.array(
+            [self.feature_weights_list[0][i] for i in self.data_interface.continuous_feature_indexes])
+        product = np.multiply(
+            (abs(x_hat - query_instance_normalized)[:, [self.data_interface.continuous_feature_indexes]]),
+            feature_weights)
         product = product.reshape(-1, product.shape[-1])
-        proximity_loss = np.sum(product, axis=1)
-        return proximity_loss
+        proximity_loss = np.sum(product,
+                                axis=1)  # Dividing by the sum of feature weights to normalize proximity loss
+        return proximity_loss / sum(feature_weights)
 
     def compute_sparsity_loss(self, cfs):
         """Compute weighted distance between two vectors."""
         sparsity_loss = np.count_nonzero(cfs - self.x1, axis=1)
-        return sparsity_loss
+        return sparsity_loss / len(
+            self.data_interface.feature_names)  # Dividing by the number of features to normalize sparsity loss
```

Comment:
should the sparsity_loss be divided by total number of features or the total number of one-hot encoded features? Good to check here--I'm not sure.

Reply:
Great point!

Reply:
It will always be the number of features since the compute_sparsity_loss is only called for members of the population which are constructed initially in the following statement:
`self.cfs = np.zeros((self.population_size, self.data_interface.number_of_features))`